cmd/cueckoo/cmd/importpr.go

```diff
@@ -183,3 +203,72 @@ func run(ctx context.Context, name string, args ...string) (string, error) {
 	}
 	return string(out), err
 }
+
+var (
+	// regexpTrailer represents a token: value pair. Uses a rough translation of
+	// the description about what a token: value pair can be from "man
+	// git-interpret-trailers". This regexp explicitly does not try to
+	//
+	// TODO: refine to be complete. Use "man git-interpret-trailers"?
+	regexpTrailer          = regexp.MustCompile(`^[-a-zA-Z]+\s*:`)
+	regexpContinuationLine = regexp.MustCompile(`^\s+`)
+)
+
+// addCloses adds the message to "Closes #pr as merged." to commit msg. It
+// respects trailers and leaves a newline at the end of the message. Like
+// git it respects the last block of trailers.
+//
+// There is probably a nice package somewhere for parsing the git commit
+// message into the constituent parts: pre-trailers, and trailers. For now
+// we brute force it based on the logic described in "man
+// git-interpret-trailers". If there are trailers we want to insert "Closes
+// #PR as merged." as the last clear line before the trailers. If there are
+// no trailers, it should be the final line in the commit message.
```

Comment:
How about using `git interpret-trailers`? It knows how to separate and parse the trailers, and it also unfolds them, so there's no need to handle continuation lines ourselves.

```
$ git log -1 --format=%B
cue/context: support internal interpreters

This introduces the hooks needed for implementing a WASM
interpreter to be used in CUE. This API, for now, only supports
internal types.

Signed-off-by: Marcel van Lohuizen <mpvl@gmail.com>
Change-Id: Id201419028ca8cbac5883809cdbd53ab7aa440cf
Reviewed-on: https://review.gerrithub.io/c/cue-lang/cue/+/550265
Reviewed-by: Roger Peppe <rogpeppe@gmail.com>
Unity-Result: CUEcueckoo <cueckoo@cuelang.org>
TryBot-Result: CUEcueckoo <cueckoo@cuelang.org>

$ git log -1 --format=%B | git interpret-trailers --only-trailers --only-input --unfold
Signed-off-by: Marcel van Lohuizen <mpvl@gmail.com>
Change-Id: Id201419028ca8cbac5883809cdbd53ab7aa440cf
Reviewed-on: https://review.gerrithub.io/c/cue-lang/cue/+/550265
Reviewed-by: Roger Peppe <rogpeppe@gmail.com>
Unity-Result: CUEcueckoo <cueckoo@cuelang.org>
TryBot-Result: CUEcueckoo <cueckoo@cuelang.org>
```

There's apparently no way to ask it to give us the "body" of the commit message, but we can hack that together by running the helper twice. With `--only-trailers --only-input --unfold` it only gives us the trailers, and with `--only-input --unfold` it gives us everything. If you trim whitespace from both, you can count the number of trailer lines and strip that number of trailing lines from the "all" output to get the body.

Then add the body, the "closes" paragraph, and the unfolded trailers. I realise this is a bit of a refactor, so I'm happy to volunteer to do it if it sounds like a good idea. I'd just feel better about maintaining that code, as we bake in fewer assumptions and there's less chance that we messed up the string handling or logic :)

Reply:
Thanks for the suggestion. Made this change.